test/unit/test_conf.ml

```diff
@@ -0,0 +1,53 @@
+let test_check_version =
+  let test name ~config ~exe expected =
+    let test_name = "compare_version: " ^ name in
+    ( test_name
+    , `Quick
+    , fun () ->
+        let got = Ocamlformat_lib.Conf.check_version ~config ~exe in
+        Alcotest.check Alcotest.(result unit string) test_name expected got
+    )
+  in
+  [ test "empty config" ~config:"" ~exe:""
+      (Error "malformed version number \"\".")
+  ; test "invalid config" ~config:"0.33..foo" ~exe:""
+      (Error "malformed version number \"0.33..foo\".")
+  ; test "invalid exe" ~config:"0.14.2" ~exe:"0.33..foo"
+      (Error
+         "expected ocamlformat version to be \"0.14.2\" but got \
+          \"0.33..foo\".")
+  ; test "unknown 'exe'" ~config:"0.14.2" ~exe:"unknown"
+      (Error
+         "expected ocamlformat version to be \"0.14.2\" but got \
+          \"unknown\". Could not determine current ocamlformat version.")
+  ; test "equal" ~config:"0.14.1" ~exe:"0.14.1" (Ok ())
+  ; test "accepted" ~config:"0.14.1" ~exe:"0.14.1-15-g273f6f6-dirty" (Ok ())
```

Comment:
This case is worrying. This is the situation that would arise when someone has a checkout around and mistakenly ends up using it. This is one of the scenarios that the version check is intended to protect against.

What is the desired use case for this new loose check? I ask because I don't understand why the existing `-no-version-check` option is insufficient. Also, I don't know why it would not work to allow some wildcards in the specified versions in order to indicate that a loose check is desired.

Reply:
My idea is that `x.y.z-foo` should be close enough to `x.y.z` to be accepted, otherwise the version number would have been `x.y.z+1` or more. Basically:
- if .ocamlformat sets `version=x.y.z-fooo`, we check that we run the exact commit build
- if .ocamlformat sets `version=x.y.z`, we check that we are running a build of the `x.y.z` "family" that might be the exact version, or some commits after

Reply:
Yes, that is the situation I'm worried about. It is not unexpected for a config file to have a constraint of the form `version=x.y.z` and failing if the binary version is something like `x.y.z-d` is precisely a goal of the version check.

I still don't understand where this is needed that `--no-version-check` is insufficient for?

Reply:
I thought this new behavior (and not requiring --no-version-check) was a good thing (I can't find anything about a need for this, I thought I read that somewhere though), but if it's not I will close this PR .